### ISSUE TITLE
Rule proposal: force `method: "POST"` on fetch calls with body

### ISSUE THREAD

**Issue by GauBen** [sindresorhus/eslint-plugin-unicorn#1989]
### Description

I sometimes forget to add `method: 'POST'` to fetch calls

<img width="465" alt="image" src="https://user-images.githubusercontent.com/48261497/203305304-19f9808f-0232-4e25-9596-1ff7bfa1e6a7.png">

### Fail

```js
fetch('http://example.com', { body: '{}' })
```

### Pass

```js
fetch('http://example.com', { method: 'POST', body: '{}' })
```

**Comment by B2o5T** [comment]
Another thing related to `fetch` - disallow `method: 'GET'` as it will be set by default

**Comment by fisker** [comment]
Should not enforce `POST`, but forbid `'GET'`(or omitted)/`'HEAD'`

**Comment by silverwind** [comment]
No HTTP method strictly forbids a body, and there are definitely examples of APIs that accept a body on GET or DELETE, with the latter being even more common.

**Comment by silverwind** [comment]
Interesting, didn't know that. I guess some implemententations may opt to not implement that step, but for a general case, it seems good to have the rule for it then. 

I do forsee that the spec may change in the future, similar to how initially a body on DELETE was forbidden by older RFCs, but newer specs now say it MAY have a body 😉.

**Comment by fisker** [comment]
`no-invalid-fetch-options`?

**Comment by sindresorhus** [comment]
> no-invalid-fetch-options

👍
